ADVANCED.md

```diff
@@ -185,6 +208,75 @@ ReactDOM.render(
 
 You can do this for [class components](https://gist.github.com/karol-majewski/befaf05af73c7cb3248b4e084ae5df71) too (Credit: [Karol Majewski](https://twitter.com/WrocTypeScript/status/1163234064343736326))
 
+### Generic components with children
+
+`children` is usually not defined as a part of the props type. Unless `children` are explicitly defined as a part of the `props` type, an attempt to use `props.children` in JSX or in the function body will fail:
+
+```tsx
+interface WrapperProps<T> {
+  item: T;
+  renderItem: (item: T) => React.ReactNode;
+}
+
+/* Property 'children' does not exist on type 'WrapperProps<T>'. */
+const Wrapper = <T extends {}>(props: WrapperProps<T>) => {
+  return (
+    <div>
+      {props.renderItem(props.item)}
+      {props.children}
+    </div>
+  );
+};
+
+/*
+Type '{ children: string; item: string; renderItem: (item: string) => string; }' is not assignable to type 'IntrinsicAttributes & WrapperProps<string>'.
+  Property 'children' does not exist on type 'IntrinsicAttributes & WrapperProps<string>'.
+*/
+
+const wrapper = (
+  <Wrapper item="test" renderItem={item => item}>
+    {test}
+  </Wrapper>
+);
+```
+
+To work around that, either add `children` to the `WrapperProps` definition (possibly narrowing down its type, as needed):
+
+```tsx
+interface WrapperProps<T> {
+  item: T;
+  renderItem: (item: T) => React.ReactNode;
+  children: string; // The component will only accept a single string child
+}
+
+const Wrapper = <T extends {}>(props: WrapperProps<T>) => {
+  return (
+    <div>
+      {props.renderItem(props.item)}
+      {props.children}
+    </div>
+  );
+};
+```
+
+or wrap the type of the props in `React.PropsWithChildren` (this is what `React.FC<>` does):
```

Comment:
It might be easier for most people to just have `children?: React.ReactNode` as the example rather than wrapping with `PropsWithChildren`, but that's a personal preference on my part.

Reply:
Me too, I always prefer typing children explicitly.

Wrapping however helps a great deal with the transition from classes to functional components: you don't have to change the props types at all if you wrap them on the component side, and it keeps noise in the PR's to the minimum.